src/m365/spo/commands/site/site-admin-remove.ts

```diff
@@ -0,0 +1,289 @@
+      if (args.options.asAdmin) {
+        await this.callActionAsAdmin(logger, args, loginNameToRemove);
+        return;
+      }
+
+      await this.callAction(logger, args, loginNameToRemove);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async callActionAsAdmin(logger: Logger, args: CommandArgs, loginNameToRemove: string): Promise<void> {
+    await logger.logToStderr('Removing site administrator as an administrator...');
+
+    const adminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
+    const siteId = await this.getSiteId(args.options.siteUrl, logger);
+    const primaryAdminLoginName = await this.getPrimaryAdminLoginNameFromAdmin(adminUrl, siteId);
+
+    if (loginNameToRemove === primaryAdminLoginName) {
+      throw 'You cannot remove the primary site collection administrator.';
+    }
+
+    const existingAdmins = await this.getSiteAdmins(adminUrl, siteId);
+    const adminsToSet = existingAdmins.filter(u => u.loginName.toLowerCase() !== loginNameToRemove.toLowerCase());
+    await this.setSiteAdminsAsAdmin(adminUrl, siteId, adminsToSet);
+  }
+
+  private async getSiteId(siteUrl: string, logger: Logger): Promise<string> {
+    const siteGraphId = await spo.getSiteId(siteUrl, logger, this.verbose);
+    const match = siteGraphId.match(/,([a-f0-9\-]{36}),/i);
+    if (!match) {
+      throw `Site with URL ${siteUrl} not found`;
+    }
+
+    return match[1];
+  }
+
+  private async getPrimaryAdminLoginNameFromAdmin(adminUrl: string, siteId: string): Promise<string> {
+    const requestOptions: CliRequestOptions = {
+      url: `${adminUrl}/_api/SPO.Tenant/sites(%27${siteId}%27)?$select=OwnerLoginName`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        'content-type': 'application/json;charset=utf-8'
+      }
+    };
+
+    const response: string = await request.get<string>(requestOptions);
+    const responseContent = JSON.parse(response);
+    return responseContent.OwnerLoginName;
+  }
+
+  private async getSiteAdmins(adminUrl: string, siteId: string): Promise<AdminUserResult[]> {
+    const requestOptions: CliRequestOptions = {
+      url: `${adminUrl}/_api/SPO.Tenant/GetSiteAdministrators?siteId=%27${siteId}%27`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        'content-type': 'application/json;charset=utf-8'
+      }
+    };
+
+    const response: string = await request.post<string>(requestOptions);
+    const responseContent: AdminResult = JSON.parse(response);
+    return responseContent.value;
+  }
+
+  private async getCorrectLoginName(options: Options): Promise<string> {
+    if (options.userId || options.userName) {
+      let requestUrl: string = `https://graph.microsoft.com/v1.0/users/${options.userId}`;
+
+      if (options.userName) {
+        requestUrl = `https://graph.microsoft.com/v1.0/users('${formatting.encodeQueryParameter(options.userName)}')`;
+      }
```

Comment:
Seems a bit strange. Usually we extract common part and add to string rest of the url based on the options passed.

so we could do something like:
```suggestion
      let requestUrl: string = `https://graph.microsoft.com/v1.0/users/`;

      if (options.userName) {
        requestUrl += formatting.encodeQueryParameter(options.userName);
      }
      else {
        reuqestUrl += options.userId
      }
```

or to an inline if and have this in a single line as well

Reply:
I changed it a little bit , in one case there is /  and in second one is  (   . Nice idea.